File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Cpp/src/BasicSupportHandler.cpp
#include "../include/BasicSupportHandler.h"
#include <iostream>

void BasicSupportHandler::handleRequest(const SupportRequest& request) {
    if (request.getType() == "basic") {
        std::cout << "BasicSupportHandler: Handling basic request.\n";
    } else if (nextHandler) {
        nextHandler->handleRequest(request);
    } else {
        std::cout << "BasicSupportHandler: No handler found for " << request.getType() << "\n";
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Java/BasicSupportHandler.java
public class BasicSupportHandler extends SupportHandler {

    @Override
    public void handleRequest(SupportRequest request) {
        if (request.getType().equalsIgnoreCase("basic")) {
            System.out.println("BasicSupportHandler: Handling basic request.");
        } else if (nextHandler != null) {
            nextHandler.handleRequest(request);
        } else {
            System.out.println("BasicSupportHandler: No handler found for " + request.getType());
        }
    }
}
